packages/tpetra/core/ext/TpetraExt_TripleMatrixMultiply_def.hpp

```diff
@@ -680,14 +809,14 @@ namespace Tpetra {
               continue; // skip explicitly stored zero values in A
 
 
-            if (Acol2PRow[l] != LO_INVALID) {
-              // mfh 27 Sep 2016: If the entry of Acol2PRow
+            if (Acol2Prow[l] != LO_INVALID) {
+              // mfh 27 Sep 2016: If the entry of Acol2Prow
               // corresponding to the current entry of A is populated, then
               // the corresponding row of P is in P_local (i.e., it lives on
               // the calling process).
 
               // Local matrix
-              size_t Pl = Teuchos::as<size_t>(Acol2PRow[l]);
+              size_t Pl = Teuchos::as<size_t>(Acol2Prow[l]);
```

Comment:
`Teuchos::as` is unnecessary here, since the cast could never overflow.

Reply:
Ah. Cut and paste! The existing code doesn't do this correctly either.

Reply:
So, good ol' C-style (size_t) casts then?

Reply:
`static_cast` avoids any warnings relating to signed-to-unsigned conversions:
```c++
const size_t Pl = static_cast<size_t> (Acol2Prow[l]);
```